src/js/binary.js

```diff
@@ -29,6 +29,109 @@ export function navMenuListener() {
     });
 }
 
+export function selectDropdown(selector, has_label) {
+    if (!selector) return;
+
+    let $select_dropdown, $list, $list_items;
+
+    function init() {
+        const is_initialized = $(selector).hasClass('select-hidden');
+        if (!is_initialized) {
+            $(selector).addClass('select-hidden')
+                .wrap('<div class="select"></div>')
+                .after('<div class="select-dropdown"></div>');
+        }
+
+        $select_dropdown = $(selector).next('div.select-dropdown');
+        $select_dropdown.text($(selector).find(':selected').text()).wrapInner('<span></span>');
+
+        $list = $select_dropdown.parent().find('.select-options');
+        if ($list.length) {
+            // empty list to repopulate
+            $list.empty();
+        } else {
+            // create list
+            $list = $('<ul />', { class: 'select-options' }).insertAfter($select_dropdown);
+        }
+
+        const optgroups = $(selector).children('optgroup');
+        if (optgroups.length) {
+            // break down group into labels with its list items
+            optgroups.each((idx, el) => {
+                const options = $(el).children();
+                const label   = $(el).attr('label');
+                appendToList(options, label);
+            });
+        } else {
+            const options = $(selector).children('option');
+            appendToList(options);
+        }
+
+        // Attach event listeners
+        $select_dropdown.off('click').on('click', (e) => {
+            e.stopPropagation();
+            // expand dropdown expand/collapse
+            const $siblings = $('.select-dropdown').not(e.target);
+            if ($siblings.hasClass('show')) {
+                $siblings.removeClass('show');
+            }
+            $(e.target).toggleClass('show');
+        });
+
+        $list_items = $list.children('li');
+        $list_items.off('click').on('click', (e) => {
+            e.stopPropagation();
+            const target = e.target;
+            if (e.target.classList.contains('disabled')) return;
+            $select_dropdown.text($(target).text()).removeClass('show').wrapInner('<span></span>');;
+
+            const selected_value = $(selector).val();
+            const dropdown_value = $(target).attr('value');
```

Comment:
why not stick to javascript? e.g. `target.getAttribute('value')`. we should either consistently use javascript or jquery, it's weird to keep switching, we are using `.classList.contains` on line 85 which is javascript